src/backend/distributed/planner/deparse_shard_query.c

```diff
@@ -90,11 +90,23 @@ RebuildQueryStrings(Query *originalQuery, List *taskList)
 			}
 		}
 
-		deparse_shard_query(query, relationId, task->anchorShardId,
-							newQueryString);
+		/*
+		 * UPDATE and DELETE queries can include subqueries, so treat them
+		 * differently.
```

Comment:
Here we can call UpdateRelationToShardNames() on UPDATE and DELETEs and then run deparse_shard_query() without calling pg_get_query_def(). However, I think this a bit confusing, because shard id used by deparse_shard_query() becomes a non-effective operation and reading this would be a bit confusing. I'm updating the comment.